org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/TestBuilderParticipant.java

```diff
@@ -47,4 +50,8 @@ public boolean isAnnotationProcessor() {
 public void processAnnotations(BuildContext[] files) {
 	PARTICIPANT.processAnnotations(files);
 }
+
+public Optional<byte[]> postProcess(BuildContext file, ByteArrayInputStream bytes) {
```

Comment:
is there a way to access the name of the to be generated class files
as org.eclipse.jdt.core.JavaCore.getGeneratedResources(IRegion, boolean)
would produce it?
in the build context i dont see it.
the caller has access to the classfile
but this one has it neither
org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.writeClassFile(ClassFile, SourceFile, boolean)
will have it

@szarnekow our test is green anyway.
thus i wonder why we need to touch the class file

Reply:
touch is needed in older versions like 2202203. so i assume the needrebuild fix also fixed that

Reply:
> is there a way to access the name of the to be generated class files as org.eclipse.jdt.core.JavaCore.getGeneratedResources(IRegion, boolean) would produce it? in the build context i dont see it. the caller has access to the classfile but this one has it neither org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.writeClassFile(ClassFile, SourceFile, boolean) will have it

With the current implementation this is not possible (as far as I know) and also not required for this issue, but #694 intends to introduce this possibility.